### PR TITLE
W7 -VietPhan - UI Test

### PR DESCRIPTION
**Pull request by at-vietphan**
#### Related issues (Optional)
Task #84 

#### Description (Required)
- [x] Write class UI test
- [x] Show dialog when login success
- [x] Show toast when invalidate

### CODE REVIEW

File: app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
+package vn.asiantech.internship;
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java]: WARNING: [TooManyStaticImports] Too many static imports may lead to messy code
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#TooManyStaticImports

File: app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+public class LoginUITest {
+
+    @SuppressWarnings("unchecked")
+    @Rule
+    public ActivityTestRule<UnitTestActivity> mActivityTestRule =
+            new ActivityTestRule(UnitTestActivity.class);
+
+    @Test
+    public void checkLengthUserName() {
+        onView(ViewMatchers.withId(R.id.edtUserName)).perform(typeText("Admi"), closeSoftKeyboard());
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'ViewMatchers.withId' due to existing static import 'android.support.test.espresso.matcher.ViewMatchers.withId'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        onView(ViewMatchers.withId(R.id.edtPassword)).perform(typeText("PASsword1"), closeSoftKeyboard());
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java]: WARNING: [AvoidDuplicateLiterals] The String literal "PASsword1" appears 4 times in this file; the first occurrence is on line 41
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

Comment (at-dev-support-01) [inline app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'ViewMatchers.withId' due to existing static import 'android.support.test.espresso.matcher.ViewMatchers.withId'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        onView(withId(R.id.btnLogin)).perform(click());
+        onView(withText(startsWith("user name must have length more than 5 and a little than 24")))
+                .inRoot(withDecorView(not(is(mActivityTestRule.getActivity().getWindow().getDecorView()))))
+                .check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void checkLeastOneCapitalLetterUserName() {
+        onView(withId(R.id.edtUserName)).perform(typeText("viet12"), closeSoftKeyboard());
+        onView(ViewMatchers.withId(R.id.edtPassword)).perform(typeText("PASsword1"), closeSoftKeyboard());
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'ViewMatchers.withId' due to existing static import 'android.support.test.espresso.matcher.ViewMatchers.withId'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        onView(withId(R.id.btnLogin)).perform(click());
+        onView(withText(startsWith("username must have least one capital letter")))
+                .inRoot(withDecorView(not(is(mActivityTestRule.getActivity().getWindow().getDecorView()))))
+                .check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void checkSpecialCharAndSpaceUserName() {
+        onView(withId(R.id.edtUserName)).perform(typeText(") Viet12"), closeSoftKeyboard());
+        onView(ViewMatchers.withId(R.id.edtPassword)).perform(typeText("PASsword1"), closeSoftKeyboard());
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'ViewMatchers.withId' due to existing static import 'android.support.test.espresso.matcher.ViewMatchers.withId'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+        onView(withId(R.id.btnLogin)).perform(click());
+        onView(withText(startsWith("username must haven't special char and white space")))
+                .inRoot(withDecorView(not(is(mActivityTestRule.getActivity().getWindow().getDecorView()))))
+                .check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void checkMostTwoNumberUserName() {
+        onView(withId(R.id.edtUserName)).perform(typeText("Viet123"), closeSoftKeyboard());
+        onView(ViewMatchers.withId(R.id.edtPassword)).perform(typeText("PASsword1"), closeSoftKeyboard());
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'ViewMatchers.withId' due to existing static import 'android.support.test.espresso.matcher.ViewMatchers.withId'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+        onView(withId(R.id.edtUserName)).perform(typeText("PASword1"), closeSoftKeyboard());
+        onView(withId(R.id.btnLogin)).perform(click());
+        onView(withText(startsWith("password must have different username")))
+                .inRoot(withDecorView(not(is(mActivityTestRule.getActivity().getWindow().getDecorView()))))
+                .check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void checkAtLeastSpecialOrNumber() {
+        onView(withId(R.id.edtUserName)).perform(typeText("Vietviet1"), closeSoftKeyboard());
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java]: WARNING: [AvoidDuplicateLiterals] The String literal "Vietviet1" appears 5 times in this file; the first occurrence is on line 90
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

File: app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java [hunk 8]
```diff
@@ -0,0 +1,135 @@
+package vn.asiantech.internship;
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java]: WARNING: [TooManyStaticImports] Too many static imports may lead to messy code
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#TooManyStaticImports

**Comment by at-vietphan** [comment]
@at-trungnguyen Please review this pull request. Thanks :+1:

File: app/src/at-vietphan/java/vn/asiantech/internship/ui/unittest/UnitTestActivity.java [hunk 9]
```diff
@@ -13,6 +14,7 @@
  * Class UnitTestActivity
  */
 public class UnitTestActivity extends AppCompatActivity implements View.OnClickListener {
+    boolean mCheckLogin = false;
```
Comment (at-trungnguyen) [inline app/src/at-vietphan/java/vn/asiantech/internship/ui/unittest/UnitTestActivity.java]: start by `is`

File: app/src/at-vietphan/java/vn/asiantech/internship/ui/unittest/UnitTestActivity.java [hunk 10]
```diff
@@ -84,6 +85,14 @@ private void showToast(String text) {
 
     @Override
     public void onClick(View view) {
+        if (mCheckLogin) {
```
Comment (at-trungnguyen) [inline app/src/at-vietphan/java/vn/asiantech/internship/ui/unittest/UnitTestActivity.java]: not enough condition, if username or password valid this case is `true` and dialog will show.

File: app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java]: WARNING: [TooManyStaticImports] Too many static imports may lead to messy code
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#TooManyStaticImports

Comment (at-dev-support-01) [inline app/src/androidTestAt-vietphanDebug/java/vn/asiantech/internship/LoginUITest.java]: WARNING: [TooManyStaticImports] Too many static imports may lead to messy code
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#TooManyStaticImports

**Pull request merged**